File: app/src/main/java/com/pluu/webtoon/ui/detail/DaumChattingFragment.java
package com.pluu.webtoon.ui.detail;

import android.annotation.SuppressLint;
import android.graphics.Rect;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.GestureDetector;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;

import com.pluu.webtoon.R;
import com.pluu.webtoon.adapter.DetailChatAdapter;
import com.pluu.webtoon.adapter.viewholder.BaseChattingViewHolder;
import com.pluu.webtoon.item.DetailView;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Daum Chatting Fragment
 * Created by PLUUSYSTEM-NEW on 2016-01-30.
 */
@SuppressLint("ValidFragment")
public class DaumChattingFragment extends BaseDetailFragment {

    private final GestureDetector gd;

    @BindView(R.id.chattingList)
    RecyclerView chattingList;

    private DetailChatAdapter adapter;
    private Unbinder bind;

    public DaumChattingFragment(GestureDetector gd) {
        this.gd = gd;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_daum_chatting, container, false);
        bind = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        initChattingSetting();
        bindListener.firstBind();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        bind.unbind();
    }

    private void initChattingSetting() {
        chattingList.setLayoutManager(new LinearLayoutManager(getContext()));

        adapter = new DetailChatAdapter(getContext()) {
            @Override
            public BaseChattingViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
                BaseChattingViewHolder holder = super.onCreateViewHolder(parent, viewType);
                holder.itemView.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        listener.childCallToggle(false);
                    }
                });
                return holder;
            }
        };
        chattingList.setAdapter(adapter);

        final int padding = getResources().getDimensionPixelSize(R.dimen.activity_horizontal_margin);
        chattingList.addOnItemTouchListener(new RecyclerView.OnItemTouchListener() {
            @Override
            public boolean onInterceptTouchEvent(RecyclerView rv, MotionEvent e) {
                return gd.onTouchEvent(e);
            }

            @Override
            public void onTouchEvent(RecyclerView rv, MotionEvent e) { }

            @Override
            public void onRequestDisallowInterceptTouchEvent(boolean disallowIntercept) { }
        });
        chattingList.addItemDecoration(new RecyclerView.ItemDecoration() {
            @Override
            public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
                super.getItemOffsets(outRect, view, parent, state);
                outRect.bottom = padding;
            }
        });
    }

    public void loadView(List<DetailView> list) {
        chattingList.setVisibility(View.VISIBLE);
        adapter.clear();
        adapter.setList(list);
        adapter.notifyDataSetChanged();
        chattingList.scrollToPosition(0);
        listener.childCallToggle(true);
    }

}

File: app/src/main/java/com/pluu/webtoon/ui/detail/DaumChattingFragment.kt
package com.pluu.webtoon.ui.detail

import android.annotation.SuppressLint
import android.graphics.Rect
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.*
import com.pluu.webtoon.R
import com.pluu.webtoon.adapter.DetailChatAdapter
import com.pluu.webtoon.adapter.viewholder.BaseChattingViewHolder
import com.pluu.webtoon.item.DetailView
import kotlinx.android.synthetic.main.fragment_daum_chatting.*

/**
 * Daum Chatting Fragment
 * Created by pluu on 2017-05-06.
 */
@SuppressLint("ValidFragment")
class DaumChattingFragment(private val gd: GestureDetector) : BaseDetailFragment() {

    private var adapter: DetailChatAdapter? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?) =
            inflater?.inflate(R.layout.fragment_daum_chatting, container, false)

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initChattingSetting()
        firstBind()
    }

    private fun initChattingSetting() {
        chattingList?.layoutManager = LinearLayoutManager(context)

        adapter = object : DetailChatAdapter(context) {
            override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseChattingViewHolder {
                val holder = super.onCreateViewHolder(parent, viewType)
                holder.itemView.setOnClickListener { _ -> listener!!.childCallToggle(false) }
                return holder
            }
        }
        chattingList?.adapter = adapter

        val padding = resources.getDimensionPixelSize(R.dimen.activity_horizontal_margin)
        chattingList?.addOnItemTouchListener(object : RecyclerView.OnItemTouchListener {
            override fun onInterceptTouchEvent(rv: RecyclerView, e: MotionEvent): Boolean {
                return gd.onTouchEvent(e)
            }

            override fun onTouchEvent(rv: RecyclerView, e: MotionEvent) {}

            override fun onRequestDisallowInterceptTouchEvent(disallowIntercept: Boolean) {}
        })
        chattingList?.addItemDecoration(object : RecyclerView.ItemDecoration() {
            override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State?) {
                super.getItemOffsets(outRect, view, parent, state)
                outRect.bottom = padding
            }
        })
    }

    override fun loadView(list: List<DetailView>) {
        chattingList?.visibility = View.VISIBLE
        adapter?.clear()
        adapter?.setList(list)
        adapter?.notifyDataSetChanged()
        chattingList?.scrollToPosition(0)
        listener?.childCallToggle(true)
    }

}
